jina/serve/runtimes/gateway/graph/topology_graph.py

```diff
@@ -57,8 +57,9 @@ def __init__(
         def leaf(self):
             return len(self.outgoing_nodes) == 0
 
-        def _update_requests(self):
+        def _update_requests_with_filter_condition(self):
             for i in range(len(self.parts_to_send)):
+                self.parts_to_send[i].docs
```

Comment:
why do u need this? is very weird. it is a code smell

Reply:
Fixed it know by overriding the deepcopy method of DataRequest